### PR TITLE
Accept leading whitespace in ics dates

### LINKED ISSUE #3403: Calendar importer errors for dates with leading whitespace
**Bug in web app**

**Describe the bug**
Importing of slightly malformed `.ics` files produces an (uncaught) error when the time entry has a leading whitespace,

**To Reproduce**
 
1. Import the following file into a calendar, note the leading space on the dates. This file is from a user:

```
BEGIN:VCALENDAR
BEGIN:VEVENT
SUMMARY:Your fancy meeting
TZID:America/New_York
DTSTART: 20211010T100000
DTEND: 20211010T120000
STATUS:CONFIRMED
ACTION: DISPLAY
CLASS:PUBLIC
METHOD:PUBLISH
UID:myuid@tutanota.com
END:VEVENT
END:VCALENDAR
```

 2. See error

**Expected behavior**
Event is imported

**Additional context**
We parse the date sliced and assume the first 4 digits represent the year. Thus, we are in year 202 here... :man_facepalming:

### PR DESCRIPTION
**Pull request by jowlo**
fix #3403

### CODE REVIEW

File: src/calendar/export/CalendarParser.js [hunk 1]
```diff
@@ -211,6 +211,7 @@ export function parseICalendar(stringData: string): ICalObject {
 	const withFoldedLines = stringData
 		.replace(/\r?\n\s/g, "")
 		.split(/\r?\n/)
+		.filter(e => e.trim()) // remove empty lines
```
Comment (johnbotris) [inline src/calendar/export/CalendarParser.js:214]: an explicit `!== ""` would make me happy

Comment (jowlo) [reply src/calendar/export/CalendarParser.js]: Happyness achieved?

Comment (johnbotris) [reply src/calendar/export/CalendarParser.js]: sehr happy

**Review by johnbotris — APPROVED** [review]

**Comment by jowlo** [comment]
## Test notes
 - [ ] Import the example file from the issue description as calendar event. The event should be imported correctly and not show an error
 - [ ] Create more `ics` files that contain tabs and other whitespaces surrounding the date and all should be imported fine.
 - [ ] regression: importing a larger `ics` file works fine as well

**Pull request merged**